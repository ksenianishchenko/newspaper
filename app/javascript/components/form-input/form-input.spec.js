import React from "react";
import { shallow } from "enzyme";
import FormInput from "./form-input";


describe("FormInput", () => {

  it("should be render correctly", () => {
    const mockFn = jest.fn();
    const tree = shallow(
      <FormInput id="3"
      name="email"
      placeholder="Your email"
      handleChange={mockFn}
      className="form-input" />
    )

    expect(tree).toMatchSnapshot();
  })

  it("should have correct props", () => {
    const mockFn = jest.fn();
    const tree = shallow(
      <FormInput id="3"
      name="email"
      placeholder="Your email"
      handleChange={mockFn}
      className="form-input" />
    )

    expect(tree.props().name).toEqual("email");
    expect(tree.props().id).toEqual("3");
    expect(tree.props().placeholder).toEqual("Your email");
    expect(tree.props().className).toEqual("form-input");
  })

  it("should call mock function on change ", () => {
    const mockFn = jest.fn();
    const tree = shallow(
      <FormInput id="3"
      name="email"
      placeholder="Your email"
      handleChange={mockFn}
      className="form-input" />
    )

    tree.simulate("change");
    expect(mockFn).toHaveBeenCalled();
  })

})
