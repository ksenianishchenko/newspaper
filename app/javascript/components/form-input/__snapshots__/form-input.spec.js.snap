// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should be render correctly 1`] = `
<input
  className="form-input"
  id="3"
  name="email"
  onChange={[MockFunction]}
  placeholder="Your email"
/>
`;
